### ISSUE TITLE
IllegalStateException: Invalid Primary Key, It should either be single field or of type composite primary key

### ISSUE THREAD

**Issue by Jordan-Nelson** [aws-amplify/amplify-android#2488]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

DataStore

### Gradle script dependencies

<details>

```groovy
// Put output below this line


```

</details>

### Environment information

com.amplifyframework:aws-datastore:2.4.1

### Describe the bug

An [issue](https://github.com/aws-amplify/amplify-flutter/issues/3084) was opened in Amplify Flutter where a user is seeing an exception when running a query with a specific schema. I have created a minimal reproduction of this issue (in flutter). The same schema/query seems to work as expected on iOS (based on my testing).

### Reproduction steps (if applicable)

1. Set up an example DS app with the schema below (can be local only, API is not required)
2. Add a "Safra" model with a "Talhao" and "Subdivisao"
3. Query "Safra" models by the "Talhao" ID using a where filter
4. Observe that the query works
5. Add a "Safra" model with a "Talhao" but no "Subdivisao"
6. Query "Safra" models by the "Talhao" ID of the new model using a where filter
7. Observe that the query throws an exception

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line

E/amplify:flutter:datastore( 5116): DataStoreException{message=Error in querying the model., cause=java.lang.IllegalStateException: Invalid Primary Key, It should either be single field or of type composite primary key Primary Key.java.lang.NullPointerException, recoverySuggestion=See attached exception for details.}
E/amplify:flutter:datastore( 5116): 	at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$query$5$com-amplifyframework-datastore-storage-sqlite-SQLiteStorageAdapter(SQLiteStorageAdapter.java:450)
E/amplify:flutter:datastore( 5116): 	at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter$$ExternalSyntheticLambda6.run(Unknown Source:10)
E/amplify:flutter:datastore( 5116): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:463)
E/amplify:flutter:datastore( 5116): 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
E/amplify:flutter:datastore( 5116): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
E/amplify:flutter:datastore( 5116): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
E/amplify:flutter:datastore( 5116): 	at java.lang.Thread.run(Thread.java:1012)
E/amplify:flutter:datastore( 5116): Caused by: java.lang.IllegalStateException: Invalid Primary Key, It should either be single field or of type composite primary key Primary Key.java.lang.NullPointerException
E/amplify:flutter:datastore( 5116): 	at com.amplifyframework.core.model.ModelIdentifier$Helper.getUniqueKey(ModelIdentifier.java:128)
E/amplify:flutter:datastore( 5116): 	at com.amplifyframework.core.model.SerializedModel$Builder.serializedData(SerializedModel.java:340)
E/amplify:flutter:datastore( 5116): 	at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.createSerializedModel(SQLiteStorageAdapter.java:900)
E/amplify:flutter:datastore( 5116): 	at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.createSerializedModel(SQLiteStorageAdapter.java:872)
E/amplify:flutter:datastore( 5116): 	at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$query$5$com-amplifyframework-datastore-storage-sqlite-SQLiteStorageAdapter(SQLiteStorageAdapter.java:444)
E/amplify:flutter:datastore( 5116): 	... 6 more
```

</details>

### GraphQL Schema

<details>

```graphql
input AMPLIFY {
  globalAuthRule: AuthRule = { allow: public }
} # FOR TESTING ONLY!
# Crop
type Safra @model {
  id: ID!
  talhaoID: ID! @index(name: "byTalhao")
  subdivisaoID: ID @index(name: "bySubdivisao")
  talhao: Talhao @belongsTo(fields: ["talhaoID"])
  subdivisao: Subdivisao @belongsTo(fields: ["subdivisaoID"])
}

# Plot
type Talhao @model {
  id: ID!
  nome: String
  Subdivisaos: [Subdivisao] @hasMany(indexName: "byTalhao", fields: ["id"])
  Safras: [Safra] @hasMany(indexName: "byTalhao", fields: ["id"])
}

# Subdivision
type Subdivisao @model {
  id: ID!
  talhaoID: ID! @index(name: "byTalhao")
  Safras: [Safra] @hasMany(indexName: "bySubdivisao", fields: ["id"])
  talhao: Talhao @belongsTo(fields: ["talhaoID"])
}



```

</details>

**Comment by tylerjroach** [comment]
Thank you for the report. I've verified the IllegalStateException happens on the query method exposed for Flutter, but even the native query method does not return data exactly as expected in this scenario. 

Will continue looking into the fix.

**Comment by ankpshah** [comment]
This has been fixed in version `2.14.11`
